File: src/WorkBC.Data/Model/JobBoard/SavedJob.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WorkBC.Data.Model.JobBoard
{
    public class SavedJob
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [ForeignKey("JobSeeker")]
        public string AspNetUserId { get; set; }

        public virtual JobSeeker JobSeeker { get; set; }

        [ForeignKey("Job")]
        public long JobId { get; set; }

        public Job Job { get; set; }

        public DateTime DateSaved { get; set; }
        public DateTime? DateDeleted { get; set; }
        public bool IsDeleted { get; set; } = false;

        [MaxLength(800)]
        public string Note { get; set; }

        public DateTime? NoteUpdatedDate { get; set; }
    }
}

File: jobboard-laravel/app/Models/SavedJob.php
<?php

namespace App\Models;

use App\Models\Concerns\SoftDeletesByFlag;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

final class SavedJob extends Model
{
    use SoftDeletesByFlag;

    protected $table = 'SavedJobs';

    protected $primaryKey = 'Id';

    protected $keyType = 'int';

    public $incrementing = true;

    public $timestamps = false;

    protected $casts = [
        'IsDeleted' => 'boolean',
        'DateSaved' => 'datetime',
        'NoteUpdatedDate' => 'datetime',
        'DateDeleted' => 'datetime',
    ];

    public function jobSeeker(): BelongsTo
    {
        return $this->belongsTo(JobSeeker::class, 'AspNetUserId', 'Id');
    }

    public function job(): BelongsTo
    {
        return $this->belongsTo(Job::class, 'JobId', 'JobId');
    }
}
